File: data/compilers_comparation/src/examples/second_example/src/func35.c
#include "second_example.h" 
array_t* func35(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array76;
   if (pCounter > 0) {
      array76 = vars->data[--pCounter];
      array76->refC++;
      DEBUG_COPY(array76->id);
   } else {
      array76 = (array_t*)malloc(sizeof(array_t));
      array76->size = 522;
      array76->refC = 1;
      array76->id = 76;
      array76->data = (unsigned int*)malloc(array76->size*sizeof(unsigned int));
      memset(array76->data, 0, array76->size*sizeof(unsigned int));
      DEBUG_NEW(array76->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array76;
   array_t* array77 = func42(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array77->id);
   free(params0.data);
   unsigned int loop24 = 0;
   unsigned int loopLimit24 = (rand()%loopsFactor)/4 + 1;
   for(; loop24 < loopLimit24; loop24++) {
      for (int i = 0; i < array76->size; i++) {
         array76->data[i]++;
      }
   }
   for (int i = 0; i < array76->size; i++) {
      if (array76->data[i] == 58) { 
         return array76;
      }
   }
   array77->refC--;
   if(array77->refC == 0) {
      free(array77->data);
      free(array77);
      DEBUG_FREE(array77->id);
   }
   return array76;
}

File: data/multilang_comparison/program_nim/src/func35.nim
proc func35*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array107: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array107 = vars.data[pCounter]
      inc array107.refC
      debugCopy(array107.id)
   else:
      array107 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array107.size = 996
      array107.refC = 1
      array107.id = 107
      array107.data = cast[ptr UncheckedArray[uint32]](alloc(array107.size * sizeof(uint32)))
      zeroMem(array107.data, array107.size * sizeof(uint32))
      debugNew(array107.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array107
   var array108: ptr ArrayT = func42(addr params0, getPath(), loopsFactor)
   debugReturn(array108.id)
   dealloc(params0.data)
   let loopLimit33 = (50) div (4) + 1
   for loop33 in 0 ..< loopLimit33:
      discard
      for i in 0 ..< array108.size:
         inc array108.data[i]
   for i in 0 ..< array107.size:
      if array107.data[i] == 90:
         return array107
   dec array107.refC
   if array107.refC == 0:
      dealloc(array107.data)
      dealloc(array107)
      debugFree(array107.id)
   return array108

